Katello/katello
# Fixes #34548 - Host collections Add & Remove modals

#### What are the changes introduced in this pull request?

![add_hc](https://user-images.githubusercontent.com/22042343/156819909-f65ab5c0-fa75-488c-b9ca-ecdf81b0c731.png)
![remove_hc](https://user-images.githubusercontent.com/22042343/156819911-9b5b276f-6ae0-45cb-a0a8-e008274fddfa.png)

* Add Host collections Add & Remove modals to Host collections overview card.
* Refactors `errorToast` in HostDetails (see relevant commit)

#### Considerations taken when implementing this change?

I was pleased to discover that there is already a single API for adding or removing host collections from a host.  The API is kinda weird (instead of saying which host collections to add or remove, you instead have to send _what you want the new list of host collections to be_) but works quite well and didn't require any changes!

Knowing that the two modals would be very similar, I combined them into one component. The use of ternaries etc. is a _bit_ higher than I'd like, but I think the tradeoffs are worth it.

Because it would cause significant complexity and is not a common use case, the tables don't have Select All available-- only Select Page. (this is ok per Maria)

#### What are the testing steps for this pull request?

* Create some host collections
* On the host collections overview, click the kebab > __Add host to collections__
* Select one or more available host collections and click Add
* Note that they appear on the card after the success toast appears
* Click the kebab >  __Remove host from collections__  - verify that the collections you just added are now showing as available to remove
* Select one or more available host collections and click Remove
* Click the kebab > __Add host to collections__ - verify that the host collection you just removed is available to add
* Make sure that any host collection at its host limit is not selectable in the table
* Change hosts using the breadcrumb switcher and verify everything still works correctly for the new host
